sdk/azcore/arm/resource_type.go

```diff
@@ -0,0 +1,107 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package arm
+
+import (
+	"fmt"
+	"strings"
+)
+
+var (
+	// SubscriptionResourceType is the ResourceType of a subscription
+	SubscriptionResourceType = NewResourceType(builtInResourceNamespace, "subscriptions")
+	// ResourceGroupResourceType is the ResourceType of a resource group
+	ResourceGroupResourceType = NewResourceType(builtInResourceNamespace, "resourceGroups")
+	// TenantResourceType is the ResourceType of a tenant
+	TenantResourceType = NewResourceType(builtInResourceNamespace, "tenants")
+	// ProviderResourceType is the ResourceType of a provider
+	ProviderResourceType = NewResourceType(builtInResourceNamespace, "providers")
+)
+
+// ResourceType represents an Azure resource type, e.g. "Microsoft.Network/virtualNetworks/subnets"
+type ResourceType struct {
+	Namespace string
+	Type      string
+	Types     []string
+
+	stringValue string
+}
+
+func (t *ResourceType) LastType() string {
+	return t.Types[len(t.Types)-1]
+}
+
+func (t *ResourceType) String() string {
+	return t.stringValue
+}
+
+// IsParentOf returns true when the receiver is the parent resource type of the child.
+func (t *ResourceType) IsParentOf(child *ResourceType) bool {
```

Comment:
I don't see a test for this, and I don't see this being used anywhere.  Can it be removed?  If not, please add a test.

Reply:
Tests added